### PR TITLE
Implement live read replication

### LINKED ISSUE #8: Live read replicas
Currently, litestream replicas a database to cold storage (`file` or `s3`). It would be nice to provide a way to replicate streams of WAL frames to other live nodes to serve as read replicas. 

This a network replication endpoint (probably `http`) as well as figuring out how to apply WAL writes to a live database.

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This pull request adds the ability for a Litestream instance to connect to another Litestream instance and stream changes in real-time. There is still a lot of work to make this production ready but that will happen incrementally in additional PRs in the future. _Also, note that this PR is a draft and still a work-in-progress._

Fixes #8

### CODE REVIEW

File: db.go [hunk 1]
```diff
@@ -1499,10 +1631,16 @@ func (db *DB) execCheckpoint(mode string) (err error) {
...
+		return db.monitorUpstream(ctx)
+	}
+	return db.monitorLocal(ctx)
+}
 
+// monitor runs in a separate goroutine and monitors the local database & WAL.
+func (db *DB) monitorLocal(ctx context.Context) error {
+	var timer *time.Timer
 	if db.MonitorDelayInterval > 0 {
 		timer := time.NewTimer(db.MonitorDelayInterval)
```
Comment (nussjustin) [inline db.go]: Just noticed this skimming through the PR: Looks like the code is accidentally shadowing `timer`

Comment (benbjohnson) [reply db.go]: Good catch! Thanks for finding that. It's fixed now. 👍

**Comment by benbjohnson** [comment]
I'm merging this initial PR in but there's a lot more features and testing to be added in smaller future PRs before releasing as `v0.4.0`.

**Comment by eazuka** [comment]
This is really awesome. can't wait for v0.4.0

**Comment by benbjohnson** [comment]
@eazuka Thanks! I added [some docs on `tip` docs site](https://tip.litestream.io/guides/read-replica/) and there's also a [litestream-read-replica-example](https://github.com/benbjohnson/litestream-read-replica-example) repository if you want to try it out.

**Pull request merged**